pkg/command/context.go

```diff
@@ -296,40 +298,45 @@ func syncContextPlugins(cmd *cobra.Command, contextType configtypes.ContextType,
 	// sort the plugins based on the plugin name
 	sort.Sort(discovery.DiscoveredSorter(plugins))
 
-	// list plugins only if listPlugins is true and there are plugins to be installed
-	if listPlugins {
-		pluginsNeedstoBeInstalled := 0
-		for idx := range plugins {
-			if plugins[idx].Status == common.PluginStatusNotInstalled || plugins[idx].Status == common.PluginStatusUpdateAvailable {
-				pluginsNeedstoBeInstalled++
-			}
-		}
-		if pluginsNeedstoBeInstalled > 0 {
-			log.Infof("The following plugins will be installed for context '%s' of contextType '%s': ", ctxName, contextType)
-			displayUninstalledPluginsContentAsTable(plugins, cmd.ErrOrStderr())
+	pluginsNeedToBeInstalled := []discovery.Discovered{}
+	for idx := range plugins {
+		if plugins[idx].Status == common.PluginStatusNotInstalled || plugins[idx].Status == common.PluginStatusUpdateAvailable {
+			pluginsNeedToBeInstalled = append(pluginsNeedToBeInstalled, plugins[idx])
 		}
 	}
 
-	err = pluginmanager.InstallDiscoveredContextPlugins(plugins)
-	if err != nil {
-		errList = append(errList, err)
+	if len(pluginsNeedToBeInstalled) == 0 {
+		log.Success("All recommended plugins are already installed.")
```

Comment:
We used to say "All recommended plugins are already installed and up-to-date" to indicate there is also nothing to update.  But if you don't like it, I don't mind getting rid of it, I'm just mentioning it in case you didn't realized.